### PR TITLE
feat: PagerDuty Integration

### PR DESCRIPTION
**Pull request by RaviHari**
Signed-off-by: Ravi Hari <ravireliable@gmail.com>
Fixes: https://github.com/argoproj/notifications-engine/issues/78

### CODE REVIEW

**Comment by harikrongali** [comment]
@RaviHari Thanks for refactoring to add serviceId as subscription

File: pkg/services/pagerduty.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+type PagerdutyOptions struct {
+	Token     string `json:"token"`
+	From      string `json:"from,omitempty"`
+	ServiceID string `json:"serviceID"`
+}
+
+func (p *PagerDutyNotification) GetTemplater(name string, f texttemplate.FuncMap) (Templater, error) {
+	Title, err := texttemplate.New(name).Funcs(f).Parse(p.Title)
```
Comment (pasha-codefresh) [inline pkg/services/pagerduty.go:26]: please change to title

File: pkg/services/pagerduty.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	if err != nil {
+		return nil, err
+	}
+	Body, err := texttemplate.New(name).Funcs(f).Parse(p.Body)
+	if err != nil {
+		return nil, err
+	}
+	Urgency, err := texttemplate.New(name).Funcs(f).Parse(p.Urgency)
```
Comment (pasha-codefresh) [inline pkg/services/pagerduty.go:34]: urgency

File: pkg/services/pagerduty.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	if err != nil {
+		return nil, err
+	}
+	PriorityID, err := texttemplate.New(name).Funcs(f).Parse(p.PriorityID)
```
Comment (pasha-codefresh) [inline pkg/services/pagerduty.go:38]: priorityId

File: docs/services/pagerduty.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Pagerduty
+
+## Parameters
+
+The Pagerduty notification service is used to create pagerduty incidents and requires specifying the following settings:
+
+* `pagerduty-token` - the pagerduty auth token
```
Comment (pasha-codefresh) [inline docs/services/pagerduty.md:7]: wdyt about rename it to pagerdutyToken to be consiste with other params

File: pkg/services/services.go [hunk 5]
```diff
@@ -8,6 +8,8 @@ import (
 	texttemplate "text/template"
 	_ "time/tzdata"
 
+	_ "github.com/golang/mock/mockgen/model"
```
Comment (pasha-codefresh) [inline pkg/services/services.go:11]: why you need this dependency here?

File: pkg/services/pagerduty.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+func NewPagerdutyService(opts PagerdutyOptions) NotificationService {
+	return &pagerdutyService{opts: opts}
+}
+
+type pagerdutyService struct {
+	opts PagerdutyOptions
+}
+
+func (p pagerdutyService) Send(notification Notification, dest Destination) error {
+	Title := notification.Pagerduty.Title
```
Comment (pasha-codefresh) [inline pkg/services/pagerduty.go:84]: same here with varibles

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]
Really great job, just few small comments

**Review by pasha-codefresh — APPROVED** [review]
Great job, thank you!

**Pull request merged**